File: ecard.web/js/OperationPointLogList.js
function bigChange(obj) {
    var smObj = document.getElementsByName("operationPointLogId");
    if (obj.checked == false) {
        for (var i = 0; i < smObj.length; i++)
            smObj[i].checked = false;
    } else {
        for (var i = 0; i < smObj.length; i++)
            smObj[i].checked = true;
    }
}
function smallChange(obj) {
    var smObj = document.getElementsByName("operationPointLogId");
    var bigObj = document.getElementById("goodslist_checkbox_all");
    if (obj.checked == true)
        bigObj.checked = true;
    else {
        b = true;
        for (var i = 0; i < smObj.length; i++) {
            if (smObj[i].checked == true)
                b = false;
        }
        if (b == true)
            bigObj.checked = false;
    }
}
var pageS = 10;
var index = 0;
var ActionName = "";
var ActionUrl = "";
var jsonData = "";
function selectInput(choose) {
    pageS = choose.value;
    submitClicks(choose);
}
//获取数据
function AjaxGetData(PageIndex, PageSize) {

    var startTime = $('#startTime').val();
    var endTime = $('#endTime').val();
    $.ajax({
        url: "/OperationPointLog/AjaxList",
        data: {
            PageIndex: PageIndex, PageSize: PageSize,  startTime: startTime,endTime: endTime
        },
        type: "post",
        //cache: false,
        dataType: "json",
        async: false,
        success: function (data) {
            $("#goodslist_checkbox_all").attr("checked", "");
            var listVal = new Array();
            listVal.push("operationPointLogId");
            listVal.push("point");
            listVal.push("account");
            listVal.push("DisplayName");
            listVal.push("Mobile");
            listVal.push("Gender");
            listVal.push("Email");
            listVal.push("remark");
            listVal.push("submitTime");
            listVal.push("boor");
            $.GetstrTrs(data, listVal, "operationPointLogId");

        },
        error: function () {
            window.location.href = "/Home/Error";
        }
    })


}
//操作
function OperatorThis(RName, RUrl) {
    ActionName = RName;
    ActionUrl = RUrl;
    switch (RName) {
        case "Edit":
            window.location.href = RUrl;
            break;
        case "Delete":
            $(".tip p").text("你确定要删除此记录吗？");
            $(".ShowHide").fadeIn(100);
            $(".tip").fadeIn(200);
            break;
        case "Deletes":
            var id = "";
            $("#tbodysNum input:checked").each(function () {
                id += $(this).val() + ",";
            })
            id = id.substring(0, id.length - 1);
            jsonData = id;
            $(".tip p").text("你确定要删除所选中记录吗？");
            $(".ShowHide").fadeIn(100);
            $(".tip").fadeIn(200);
            break;
    }
}

File: ecard.mvc/Models/OperationPointLogs/OperationPointLogList.cs
using Ecard.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Ecard.Mvc.Models.OperationPointLogs
{
    public class OperationPointLogList
    {
        private readonly OperationPointLogModel _innerObject;
        [NoRender]
        public OperationPointLogModel InnerObject
        {
            get { return _innerObject; }
        }

        public OperationPointLogList()
        {
            _innerObject = new OperationPointLogModel();
        }

        public OperationPointLogList(OperationPointLogModel innerObject)
        {
            _innerObject = innerObject;
        }
        [NoRender]
        public int operationPointLogId { get { return InnerObject.operationPointLogId; } }

        public decimal point { get { return InnerObject.point; } }

        public string account { get { return InnerObject.account; } }
        public string DisplayName { get { return InnerObject.DisplayName; } }
        /// <summary>
        /// 手机号码
        /// </summary>
        public string Mobile { get { return InnerObject.Mobile; } }
        /// <summary>
        /// 性别
        /// </summary>
        public string Gender { get { return InnerObject.Gender == 1 ? "男" : "女"; } }
        /// <summary>
        /// 电子邮箱
        /// </summary>
        public string Email { get { return InnerObject.Email; } }

        public string remark { get { return InnerObject.remark; } }

        public string submitTime { get { return InnerObject.submitTime.ToString("yyyy-MM-dd"); } }



        [NoRender]
        public string boor { get; set; }
    }
}
